### ISSUE TITLE
Flaky inttest `check-addons`

### ISSUE THREAD

**Issue by twz123** [k0sproject/k0s#3045]
The `check-addons` inttest is failing quite often nowadays. The error is

> can't install loadedChart `echo-server`: unable to build kubernetes objects from release manifest: Get "https://localhost:6443/openapi/v2?timeout=32s": stream error: stream ID xx; INTERNAL_ERROR; received from peer

This seems to be an error that helm is getting from the API server. The error message indicates that something has gone thoroughly wrong on the server side?!?

<details>
<summary>
Logs of a failed test
</summary>

```text
=== RUN   TestAddonsSuite/TestHelmBasedAddons
    footloosesuite.go:976: waiting for kube api to start on node controller0
    footloosesuite.go:980: kube-client error: Process exited with status 1
    footloosesuite.go:1001: kube api up-and-running, version: v1.27.1+k0s
    footloosesuite.go:909: waiting to see worker0 ready in kube API
    footloosesuite.go:913: worker0 is ready in API
    addons_test.go:115: waiting to see test-addon release ready in kube API, generation 1
    addons_test.go:151: found test addon release: k0s-addon-chart-test-addon
    addons_test.go:115: waiting to see oci-addon release ready in kube API, generation 1
    addons_test.go:149: 
        	Error Trace:	/home/runner/work/k0s/k0s/inttest/addons/addons_test.go:149
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/wait.go:109
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/wait.go:154
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/wait.go:207
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/poll.go:260
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/poll.go:175
        	            				/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.27.1/pkg/util/wait/poll.go:157
        	            				/home/runner/work/k0s/k0s/inttest/addons/addons_test.go:126
        	            				/home/runner/work/k0s/k0s/inttest/addons/addons_test.go:55
        	Error:      	Should be empty, but was can't install loadedChart `echo-server`: unable to build kubernetes objects from release manifest: Get "https://localhost:6443/openapi/v2?timeout=32s": stream error: stream ID 63; INTERNAL_ERROR; received from peer
        	Test:       	TestAddonsSuite/TestHelmBasedAddons
```
</details>

**Comment by twz123** [comment]
Wondering if that might be a leftover from a previous attempt that encountered a transient failure? The code looks like, once set, the error is never cleared again...

https://github.com/k0sproject/k0s/blob/a0efc68177420714bff12c1835b29f0ed3e8a6e8/pkg/component/controller/extensions_controller.go#L290-L308

**Comment by twz123** [comment]
Turns out the root cause is kubernetes/kubernetes#117363. Anyhow, I still think the the error won't get cleared after a successful reconciliation, and this is an issue in k0s itself, too.

**Comment by twz123** [comment]
I think all should be fixed by now. The upstream fix is part of k8s 1.27.2, if I'm not mistaken.
